### ISSUE TITLE
Unable to set a breakpoint in a back-piped function

### ISSUE THREAD

**Issue by jbeeko** [dotnet/fsharp#11668]
In version 16.9.x it was possible to set a breakpoint on the lines of an anonymous  function passed to a back pipe. In version 16.10.x this is no longer possible. The breakpoint is always set at the function level.

v16.10.0
<img width="332" alt="image" src="https://user-images.githubusercontent.com/4968704/121566096-050d3980-c9d2-11eb-870f-dcc9f8dbf3a8.png">

v16.9.1
<img width="475" alt="image" src="https://user-images.githubusercontent.com/4968704/121565764-a34ccf80-c9d1-11eb-8f77-94be3ae011e3.png">

**Repro steps**
Start VisualSudio v16.10.x past the code into a .fs file and attempt to set a break point on the line `printfn "line1"`. 
```
let bodyRunner  body =
    printfn "running"
    body ()

let bodyWrapper =
    bodyRunner <| fun _ ->
        printfn "line1"
        printfn "line2"
```

**Expected behavior**
Be able to set a break point on the line

**Actual behavior**
Breakpoint set at function level. 

**Known workarounds**
None

**Environment**

* Windows 10 Pro 20H2
* .NET Runtime .NET 5.0
* Fails on v16.10.0 & v16.10.1.

**Comment by cartermp** [comment]
I debugged this and confirmed there's a regression in the compiler here. One of the several PRs we took regressed on range generation for lambda expressions. The range of the body for the lambda only includes the argument(s); it doesn't have the body of the lambda anymore. Since that range is missing, the breakpoint resolution service ends up missing where you clicked and falls back to an outer scope.

**Comment by cartermp** [comment]
Haven't dived into the history much myself, but @dsyme @auduchinok and @nojaf were the last folks who changed the parser between 16.9 and 16.10. Maybe this rings a bell?

See here: https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQ2wAuYarALa8AFigDmYAPoAPMAF4wACgCUAHQBOa5Ju0DhPMckkAeAD6UArsmlgAtAD5dNiGAAOGlD3g21kAJ6wvH6QIAC%2BQA

The range for the lambda is correct in the fantomas web UI, but the range is incorrect when debugging the expression. So I suspect this is also an issue in Rider, VSCode, and anything else that would rely on the range.

**Comment by nojaf** [comment]
Nothing comes to find for me. What timeframe are we looking at here?

**Comment by jbeeko** [comment]
I'll note that this prevents breakpoints in any of the "non-builder" style expecto tests. So it has a significant impact on those with a library of such tests.

**Comment by dsyme** [comment]
In `main` I only repro this with Release mode code.

Curiously it's changed 

* Previously, no debug points on `bodyRunner` in release code (because it gets inlined?)
* Now, no debug points on the sequentials of the lambda (because inlining is happening differently after recent lambda optimizations?)

Currently, when we inline code, we remove debug ranges.  This prevents jumping into library code unexpectedly in the middle of library code.  There have been requests to change this and maintain the debug ranges - for example, when doing intra-assembly inlining.  I can see the value in this.

**Comment by dsyme** [comment]
Fixed in #11981
